target/linux/ipq40xx/patches-5.15/421-firmware-qcom-scm-disable-SDI-on-Google-WiFi.patch

```diff
@@ -0,0 +1,32 @@
+From 531c86fed172bcaef6b6eb6e92aa1ce42ff56b88 Mon Sep 17 00:00:00 2001
+From: Brian Norris <computersforpeace@gmail.com>
+Date: Fri, 28 Jul 2023 12:02:19 +0200
+Subject: [PATCH 2/3] firmware: qcom: scm: disable SDI on Google WiFi
+
+Google WiFi seems to have SDI (Secure Debug Image) enabled by default which
+prevents normal reboot from working causing the board to just hang after
+reboot is called.
+
+So lets disable SDI before rebooting on Google WiFi boards.
```

Comment:
What happens when the HW watchdog triggers a reboot?
Maybe we have a pre warning interrupt in the watchdog driver we can also hook into.

I think for OpenWrt this fix is fine and it is ok if the watchdog triggers a hang. It would be nice to handle this too.

Reply:
We can just move the SDI disable to the SCM probe(This is what Brian is doing currently) and then any WDT trigger should end up being a normal reset and not hang in debug mode.
Since SCM is one of first things being probed it should almost eliminate any chance of SDI ever being left enabled and WDT kicking in.

Reply:
Moved it back to probe time, it should be fine now

Reply:
When you disable it in the probe() function it is fine.